blur.rb

```diff
@@ -0,0 +1,124 @@
+class Image
+
+
+  def initialize(data)
+    @pixels = []
+    construct_pixels(data)
+    @height = @pixels.count - 1
+    @width  = @pixels[0].count - 1
+  end
+  
+  
+  # Convenience method for outputing the image as a nested array of 1's and 0's
+  def show
+    buffer = []
+    @pixels.each do |row|
+      temp = []
+      row.each do |pixel|
+        temp << pixel.value
+      end
+      buffer << temp
+    end
+    pp buffer
+  end
+  
+  
+  def blur(dist)
+    @pixels.each_with_index do |pixel_row, y|
+      pixel_row.each_with_index do |pixel, x|
+        if pixel.one? && !pixel.changed
+          # Ripple is the mama function here. See comments below.
+          ripple(x, y, dist)
+        end
+      end
+    end
+  end
+  
+private
+  
+  
+  def ripple(x, y, dist)
+    # There are four directions: up, down, left, right. We simply need a way to
+    # define each direction. For instance, the first object in directions below 
+    # is the object for 'up'. As you can see, the x coordinate stays the same
+    # while the y value increases (subtraction is going 'up' in this example).
+    # The four below are simply ways to define a direction in the coordinate plane.
+    directions = [
+      {x: x, y: Proc.new { |i| y - (i + 1) }},
+      {x: x, y: Proc.new { |i| y + (i + 1) }},
+      {x: Proc.new { |i| x + (i + 1) }, y: y},
+      {x: Proc.new { |i| x - (i + 1) }, y: y}
+    ]
+    directions.each do |direction|
+      dist.times do |i|
```

Comment:
I like this thought of looping through the distance, it's very close to one of the two approaches we push people towards.